src/type_.rs

```diff
@@ -120,10 +125,28 @@ impl<'gcc, 'tcx> BaseTypeMethods<'tcx> for CodegenCx<'gcc, 'tcx> {
         self.isize_type
     }
 
+    #[cfg(feature = "master")]
+    fn type_f16(&self) -> Type<'gcc> {
+        if self.supports_f16_type {
+            return self.context.new_c_type(CType::Float16);
+        }
+        unimplemented!("f16")
+    }
+
+    #[cfg(not(feature = "master"))]
     fn type_f16(&self) -> Type<'gcc> {
-        unimplemented!("f16_f128")
+        unimplemented!("f16")
+    }
+
+    #[cfg(feature = "master")]
+    fn type_f32(&self) -> Type<'gcc> {
+        if self.supports_f32_type {
+            return self.context.new_c_type(CType::Float32);
+        }
+        self.float_type
     }
 
+    #[cfg(not(feature = "master"))]
     fn type_f32(&self) -> Type<'gcc> {
         self.float_type
     }
```

Comment:
Would it work to do the following?
```suggestion
    fn type_f32(&self) -> Type<'gcc> {
        #[cfg(feature = "master")]
        if self.supports_f32_type {
            return self.context.new_c_type(CType::Float32);
        }
        self.float_type
    }
```